DigitalLearningSolutions.Web.Tests/Controllers/LearningMenu/LearningMenuControllerTests.cs

```diff
@@ -263,6 +269,145 @@ public void Sections_should_StartOrUpdate_course_sessions()
+        }
+
+        [Test]
+        public void Sections_should_not_UpdateProgress_course_sessions_if_invalid_section()
+        {
+            // Given
+            A.CallTo(() => sectionContentService.GetSectionContent(CustomisationId, CandidateId, SectionId))
+                .Returns(null);
+
+            // When
+            controller.Section(CustomisationId, SectionId);
+
+            // Then
+            A.CallTo(() => courseContentService.UpdateProgress(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void Sections_should_UpdateProgress_course_sessions_if_unable_to_enrol()
+        {
+            // Given
+            var defaultSectionContent = SectionContentHelper.CreateDefaultSectionContent();
+            A.CallTo(() => sectionContentService.GetSectionContent(CustomisationId, CandidateId, SectionId))
+                .Returns(defaultSectionContent);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId))
+                .Returns(null);
+
+            // When
+            controller.Section(CustomisationId, SectionId);
+
+            // Then
+            A.CallTo(() => courseContentService.UpdateProgress(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void Sections_should_render_view()
+        {
+            // Given
+            const int progressId = 299;
+            var defaultSectionContent = SectionContentHelper.CreateDefaultSectionContent();
+            A.CallTo(() => sectionContentService.GetSectionContent(CustomisationId, CandidateId, SectionId))
+                .Returns(defaultSectionContent);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId)).
+                Returns(progressId);
+
+            // When
+            var result = controller.Section(CustomisationId, SectionId);
+
+            // Then
+            var expectedModel = new SectionContentViewModel(defaultSectionContent, CustomisationId);
+            result.Should().BeViewResult()
+                .Model.Should().BeEquivalentTo(expectedModel);
+        }
+
+        [Test]
+        public void Sections_should_404_if_section_not_found()
+        {
+            // Given
+            A.CallTo(() => sectionContentService.GetSectionContent(CustomisationId, CandidateId, SectionId))
+                .Returns(null);
+
+            // When
+            var result = controller.Section(CustomisationId, SectionId);
+
+            // Then
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(A<int>._, A<int>._, A<int>._)).MustNotHaveHappened();
+            result.Should()
+                .BeRedirectToActionResult()
+                .WithControllerName("LearningSolutions")
+                .WithActionName("StatusCode")
+                .WithRouteValue("code", 404);
+        }
+
+        [Test]
+        public void Sections_should_500_if_failed_to_enrol()

```

Comment:
We've changed this to 404 on the tutorial and initial menu pages. I realised this could happen in a few "expected" cases, e.g. if the user centre doesn't match the customisation centre. In that case I think a 404 is more suitable 🙂